tests/unit/core/model/module/test_segmentation.py

```diff
@@ -0,0 +1,81 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Unit tests for segmentation model module."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+from unittest.mock import MagicMock
+
+import pytest
+import torch
+from omegaconf import OmegaConf
+from otx.core.data.entity.base import ImageInfo
+from otx.core.data.entity.segmentation import SegBatchDataEntity, SegBatchPredEntity
+from otx.core.model.entity.segmentation import MMSegCompatibleModel
+from otx.core.model.module.segmentation import OTXSegmentationLitModule
+
+if TYPE_CHECKING:
+    from omegaconf.dictconfig import DictConfig
+
+
+class MockMetric(torch.nn.Module):
+    def update(*args, **kwargs) -> None:
+        pass
+
+class MockModel(torch.nn.Module):
+    def __init__(self, input_dict):
+        self.input_dict = input_dict
+
+    def __call__(self, *args, **kwargs) -> SegBatchPredEntity:
+        return SegBatchPredEntity(**self.input_dict, scores=[])
+
+
+class TestOTXSegmentationModel:
+    @pytest.fixture()
+    def config(self) -> DictConfig:
+        return OmegaConf.load("src/otx/recipe/segmentation/litehrnet_18.yaml")
+
+    @pytest.fixture()
+    def input_dict(self) -> dict:
+        img_size = (240,240)
+        return {"batch_size": 1,
+                "images": [torch.rand(img_size)],
+                "imgs_info": [ImageInfo(0, img_size, img_size, img_size, img_size)],
+                "masks": [torch.rand(img_size)],
+                     }
+
+    @pytest.fixture()
+    def data_inputs(self, input_dict) -> SegBatchDataEntity:
+        return SegBatchDataEntity(**input_dict)
+
+    @pytest.fixture()
+    def model(self, config, input_dict) -> OTXSegmentationLitModule:
+        otx_model = MMSegCompatibleModel(config.model.otx_model.config)
+        model = OTXSegmentationLitModule(otx_model, MagicMock, MagicMock, False)
+        model.model = MockModel(input_dict)
+        model.val_metric = MockMetric()
+        model.test_metric = MockMetric()
```

Comment:
I just wanna you to follow the mocking best practice using `MagicMock` rather than mannually defining the mock class such as
```suggestion
from torchmetrics.metric import Metric
...

    def model(self, config, input_dict, mocker) -> OTXSegmentationLitModule:
        otx_model = mocker.MagicMock(MMSegCompatibleModel)
        model = OTXSegmentationLitModule(otx_model, MagicMock, MagicMock, False)
        model.val_metric = mocker.MagicMock(spec=Metric)
        model.test_metric = mocker.MagicMock(spec=Metric)
```

Reply:
Done [77733bb](https://github.com/openvinotoolkit/training_extensions/pull/2715/commits/77733bb3d29399f1d20d273bb3b1011e7c2defff)